docs/guides/feature-flags.rst

```diff
@@ -0,0 +1,59 @@
+Feature Flags
+=============
+
+Read the Docs offers some additional flag settings which can be only be configured by the site admin.
+These are optional settings and you might not need it for every project.
+By default, these flags are disabled for every project.
+A seperate request can be made by opening an issue on our `github`_ to enable
+or disable one or more of these featured flags for a particular project.
+
+.. _github: https://github.com/rtfd/readthedocs.org
+
+Available Flags
+---------------
```

Comment:
I think we could use a rst role to keep the list updated, something like https://github.com/rtfd/readthedocs.org/pull/4451#discussion_r207321274 https://github.com/rtfd/readthedocs.org/pull/4451/files#diff-d4de2c39f7f56cdad5a95921ae1398ec

Reply:
@stsewd 
I didn't get it.
Do you want to render the whole list (with description) with rst role? or just the description? or any other thing?

Reply:
List and description

Reply:
@stsewd 
I have tried it and i suppose that the function for "role" is working fine. However I am not able to get the output at the docs.
Can you please help me in this. I am unable to find the mistakes.
The output I'm getting is 

**featureflags:**

Reply:
@stsewd 
It was a wonderful idea. I managed to implement it successfully and now the flags with their description are rendering as desired automatically.
Here is the image of the feature flags docs page - [image](https://i.ibb.co/ynTt3ns/screencapture-file-home-pikachu-Desktop-rtd-readthedocs-org-docs-build-html-guides-feature-flags-html-2019-01-04-15-32-11.png)

Reply:
I love adding features like this to our docs! :100: 

I would suggest a change here: instead of using a role to put all of the different feature flags in a single block, this should mimic what the django setting role does.

Also, if you were to implement a feature like this normally, you'd use a `directive`, not a `role`. I do think `role` would be preferred here if you were to match how we are using the django setting `role`. It allows us to put more commentary around the output.

Reply:
@agjohnson 
I am a little confused.
By the example of django setting role, did you mean to say to use `nodes.literal` instead of `nodes.definition_list` ?

Reply:
I don't think we want to fully automate things, there might be some feature flags we don't want documented or don't need documented.

Reply:
Thank you for the explanation.
I have followed your advice and changed the pattern of the role similar to `djangosettings role`.